homepage-dashboard: 1.3.2 -> 1.4.6

jnsgruk opened the pull request:
## Things done

Bumps to the latest [upstream 1.4.3 release](https://github.com/gethomepage/homepage/releases/tag/v1.4.3) using the maintainer script, which needed a quick fix along the way!

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jnsgruk committed:
homepage-dashboard: fix update script

The update script had begun failing due to changes
in how `pnpm` and `nodejs` are packaged in
`unstable`.

This change ensures the correct tools are present
in the environment, such that the update script
can succeed.

jnsgruk committed:
homepage-dashboard: 1.3.2 -> 1.4.3

jnsgruk commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 430906`
Commit: `8628f797b96d623cfadda6d6c93c13eb5e85c12d`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>homepage-dashboard</li>
  </ul>
</details>

SuperSandro2000 reviewed (approved):
LGTM

jnsgruk commented:
Hmm, this is broken. The nixos module is writing the config files in the right place, but the app isn't reading them, hence the failing test. Converting to draft.

newAM commented:
I took a look, I'm stumped.

- it's not the systemd hardening, I removed all of that and the problem persisted
- I think the application is reading `settings.yaml`, if I remove `HOMEPAGE_CONFIG_DIR` then it writes a log message about missing settings

jnsgruk committed:
nixos/homepage-dashboard: add support for `proxmox` config

The upstream project recently added support for
a new configuration file named `proxmox.yaml`.

Without adding this, the server can fail to start
since it tries to create the file `proxmox.yaml`
but fails because `/etc` is read-only.

See https://gethomepage.dev/configs/proxmox/ for
details of the feature, which is handled here
similiarly to docker/kubernetes configs.

jnsgruk committed:
nixos/tests/homepage-dashboard: fix failing tests

This commit adjusts the test suite slightly
to request a more "static" asset to ensure that
the configuration is reflected.

Rather than request the full page, which requires
Javascript to be accurate, we request from the
API the configured bookmarks and the site
manifest which contains the configured title.

jnsgruk commented:
@SuperSandro2000 would you mind taking another look to check you're still happy? Tests should now be fixed!

jnsgruk commented:
@ofborg test homepage-dashboard

jnsgruk committed:
homepage-dashboard: 1.4.3 -> 1.4.6

jnsgruk commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 430906`
Commit: `75ccd1f28868e020ff6d48449380556792d435f8`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>homepage-dashboard</li>
  </ul>
</details>
